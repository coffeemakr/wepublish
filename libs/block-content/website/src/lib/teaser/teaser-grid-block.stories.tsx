import {css} from '@emotion/react'
import {Meta} from '@storybook/react'
import {TeaserGridBlock as TeaserGridBlockType} from '@wepublish/website/api'
import {TeaserGridBlock} from './teaser-grid-block'
import {articleTeaser, pageTeaser, customTeaser} from '@wepublish/testing/fixtures/graphql'

export default {
  component: TeaserGridBlock,
  title: 'Blocks/Teaser Grid'
} as Meta

const flexTeaser = {
  __typename: 'TeaserGridBlock',
  teasers: [articleTeaser, pageTeaser, customTeaser],
  numColumns: 3
} as TeaserGridBlockType

export const OneColumn = {
  args: {
    ...flexTeaser,
    numColumns: 1
  }
}

export const ThreeColumns = {
  args: {
    ...flexTeaser
  }
}

export const WithShowLoad = {
  args: {
    ...flexTeaser,
    showLead: true
  }
}

export const WithClassName = {
  args: {
    ...flexTeaser,
    className: 'extra-classname'
  }
}

export const WithEmotion = {
  args: {
    ...flexTeaser,
    css: css`
      background-color: #eee;
    `
  }
}
