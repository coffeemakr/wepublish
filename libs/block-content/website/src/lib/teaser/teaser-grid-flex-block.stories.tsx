import {css} from '@emotion/react'
import {Meta} from '@storybook/react'
import {TeaserGridFlexBlock} from './teaser-grid-flex-block'
import {flexTeaser} from '@wepublish/testing/fixtures/graphql'

export default {
  component: TeaserGridFlexBlock,
  title: 'Blocks/Teaser Grid Flex'
} as Meta

export const Default = {
  args: {
    ...flexTeaser
  }
}

export const WithShowLoad = {
  args: {
    ...flexTeaser,
    showLead: true
  }
}

export const WithClassName = {
  args: {
    ...flexTeaser,
    className: 'extra-classname'
  }
}

export const WithEmotion = {
  args: {
    ...flexTeaser,
    css: css`
      background-color: #eee;
    `
  }
}
