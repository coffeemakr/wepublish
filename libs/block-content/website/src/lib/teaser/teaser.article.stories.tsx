import {Meta} from '@storybook/react'
import {Teaser} from './teaser'
import {articleTeaser} from '@wepublish/testing/fixtures/graphql'

export default {
  component: Teaser,
  title: 'Blocks/Teaser/Article'
} as Meta

export const Default = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: articleTeaser
  }
}

export const WithShowLead = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: articleTeaser,
    showLead: true
  }
}

export const WithoutLead = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: {
      ...articleTeaser,
      lead: null
    },
    showLead: true
  }
}

export const WithoutLeadWithoutBlock = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: {
      ...articleTeaser,
      lead: null,
      article: {
        ...articleTeaser.article,
        blocks: []
      }
    },
    showLead: true
  }
}

export const WithoutPreTitle = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: {
      ...articleTeaser,
      preTitle: null
    }
  }
}

export const WithoutTitle = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: {
      ...articleTeaser,
      title: null
    }
  }
}

export const WithoutImage = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: {
      ...articleTeaser,
      image: null
    }
  }
}

export const WithoutImageWithoutBlock = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: {
      ...articleTeaser,
      image: null,
      article: {
        ...articleTeaser.article,
        blocks: []
      }
    }
  }
}
