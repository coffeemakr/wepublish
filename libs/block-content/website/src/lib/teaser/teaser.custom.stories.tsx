import {Meta} from '@storybook/react'
import {customTeaser} from '@wepublish/testing/fixtures/graphql'
import {Teaser} from './teaser'

export default {
  component: Teaser,
  title: 'Blocks/Teaser/Custom'
} as Meta

export const Default = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: customTeaser
  }
}

export const WithShowLead = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: customTeaser,
    showLead: true
  }
}

export const WithoutLead = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: {
      ...customTeaser,
      lead: null
    },
    showLead: true
  }
}

export const WithoutPreTitle = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: {
      ...customTeaser,
      preTitle: null
    }
  }
}

export const WithoutTitle = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: {
      ...customTeaser,
      title: null
    }
  }
}

export const WithoutImage = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: {
      ...customTeaser,
      image: null
    }
  }
}
