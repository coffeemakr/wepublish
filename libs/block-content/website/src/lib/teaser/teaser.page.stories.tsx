import {Meta} from '@storybook/react'
import {image} from '@wepublish/testing/fixtures/graphql'
import {PageTeaser} from '@wepublish/website/api'
import {Teaser} from './teaser'

export default {
  component: Teaser,
  title: 'Blocks/Teaser/Page'
} as Meta

const pageTeaser = {
  __typename: 'PageTeaser',
  style: 'DEFAULT',
  image,
  preTitle: 'Pre Title',
  title: 'Title',
  lead: 'Lead',
  page: {
    id: 'cl95fumlq261901phgrctx4mz',
    title: 'Title on the page',
    description: 'Description on the page',
    url: 'https://example.com',
    blocks: [
      {
        __typename: 'TitleBlock',
        title: 'Title from block',
        lead: 'Lead from block'
      },
      {
        __typename: 'ImageBlock',
        caption: null,
        image
      }
    ]
  }
} as PageTeaser

export const Default = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: pageTeaser
  }
}

export const WithShowLead = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: pageTeaser,
    showLead: true
  }
}

export const WithoutLead = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: {
      ...pageTeaser,
      lead: null
    },
    showLead: true
  }
}

export const WithoutLeadWithoutBlock = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: {
      ...pageTeaser,
      lead: null,
      page: {
        ...pageTeaser.page,
        blocks: []
      }
    },
    showLead: true
  }
}

export const WithoutPreTitle = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: {
      ...pageTeaser,
      preTitle: null
    }
  }
}

export const WithoutTitle = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: {
      ...pageTeaser,
      title: null
    }
  }
}

export const WithoutImage = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: {
      ...pageTeaser,
      image: null
    }
  }
}

export const WithoutImageWithoutBlock = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: {
      ...pageTeaser,
      image: null,
      Page: {
        ...pageTeaser.page,
        blocks: []
      }
    }
  }
}
