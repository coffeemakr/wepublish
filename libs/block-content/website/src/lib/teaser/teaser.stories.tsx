import {css} from '@emotion/react'
import {Meta} from '@storybook/react'
import {image} from '@wepublish/testing/fixtures/graphql'
import {CustomTeaser} from '@wepublish/website/api'
import {Teaser} from './teaser'

export default {
  component: Teaser,
  title: 'Blocks/Teaser'
} as Meta

const customTeaser = {
  style: 'DEFAULT',
  image,
  preTitle: 'preTitle',
  title: 'Teambesprechung vom 23.05.',
  lead: 'Lead',
  contentUrl: 'https://example.com',
  properties: [],
  __typename: 'CustomTeaser'
} as CustomTeaser

export const WithClassName = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: customTeaser,
    className: 'extra-classname'
  }
}

export const WithEmotion = {
  args: {
    alignment: {
      x: 0,
      y: 0,
      w: 3,
      h: 4,
      __typename: 'FlexAlignment'
    },
    teaser: customTeaser,
    css: css`
      background-color: #eee;
    `
  }
}
