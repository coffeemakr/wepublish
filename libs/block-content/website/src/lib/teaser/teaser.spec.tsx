import {runStorybookTests} from '@wepublish/testing'
import * as articleStories from './teaser.article.stories'
import * as customStories from './teaser.custom.stories'
import * as eventStories from './teaser.event.stories'
import * as pageStories from './teaser.page.stories'
import * as peerArticleStories from './teaser.peer-article.stories'
import * as stories from './teaser.stories'

describe('Teaser', () => {
  runStorybookTests(stories)

  describe('Custom', () => {
    runStorybookTests(customStories)
  })

  describe('Article', () => {
    runStorybookTests(articleStories)
  })

  describe('PeerArticle', () => {
    runStorybookTests(peerArticleStories)
  })

  describe('Page', () => {
    runStorybookTests(pageStories)
  })

  describe('Event', () => {
    runStorybookTests(eventStories)
  })
})
