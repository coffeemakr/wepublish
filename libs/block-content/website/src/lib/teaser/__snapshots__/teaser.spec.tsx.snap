// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teaser Article should render Default 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Teaser Article should render WithLightStyle 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -34,15 +34,10 @@
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title: Title
            </h1>
          </header>
-         <p
-           class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
-         >
-           Lead
-         </p>
          <div
            class="css-hg1qbe-TeaserAuthors e18of27r0"
          >
            Foobar, Foobar
          </div>"
`;

exports[`Teaser Article should render WithoutAuthors 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -39,14 +39,9 @@
          <p
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Lead
          </p>
-         <div
-           class="css-hg1qbe-TeaserAuthors e18of27r0"
-         >
-           Foobar, Foobar
-         </div>
        </div>
      </a>
    </article>
  </DocumentFragment>"
`;

exports[`Teaser Article should render WithoutBlockWithLightStyle 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -34,15 +34,10 @@
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title: Title
            </h1>
          </header>
-         <p
-           class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
-         >
-           Lead
-         </p>
          <div
            class="css-hg1qbe-TeaserAuthors e18of27r0"
          >
            Foobar, Foobar
          </div>"
`;

exports[`Teaser Article should render WithoutDate 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -19,16 +19,10 @@
  https://unsplash.it/500/500 500w,
  https://unsplash.it/300/300 300w,
  https://unsplash.it/200/200 200w"
            title="An image title"
          />
-         <time
-           class="css-3uu3f-TeaserDate e18of27r1"
-           datetime="2023-01-01T00:00:00.000Z"
-         >
-           01.01.2023
-         </time>
          <header
            class="css-4se2pd-TeaserTitle e18of27r2"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root""
`;

exports[`Teaser Article should render WithoutImage 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Teaser Article should render WithoutImageWithoutBlock 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -9,20 +9,10 @@
        href="https://example.com"
      >
        <div
          class="css-1xecuhw-TeaserInnerWrapper e18of27r3"
        >
-         <img
-           alt="An image description"
-           class="css-17etez2-ImageWrapper-Teaser e1shv2yg0"
-           loading="lazy"
-           srcset="https://unsplash.it/800/800 800w,
- https://unsplash.it/500/500 500w,
- https://unsplash.it/300/300 300w,
- https://unsplash.it/200/200 200w"
-           title="An image title"
-         />
          <time
            class="css-3uu3f-TeaserDate e18of27r1"
            datetime="2023-01-01T00:00:00.000Z"
          >
            01.01.2023"
`;

exports[`Teaser Article should render WithoutLead 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -37,11 +37,11 @@
            </h1>
          </header>
          <p
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
-           Lead
+           Lead from block
          </p>
          <div
            class="css-hg1qbe-TeaserAuthors e18of27r0"
          >
            Foobar, Foobar"
`;

exports[`Teaser Article should render WithoutPreTitle 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -31,11 +31,11 @@
            class="css-4se2pd-TeaserTitle e18of27r2"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
-             Pre Title: Title
+             Pre Title on the article: Title
            </h1>
          </header>
          <p
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >"
`;

exports[`Teaser Article should render WithoutTitle 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -31,11 +31,11 @@
            class="css-4se2pd-TeaserTitle e18of27r2"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
-             Pre Title: Title
+             Pre Title: Title from block
            </h1>
          </header>
          <p
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >"
`;

exports[`Teaser Custom should render Default 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Teaser Custom should render WithLightStyle 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -28,14 +28,9 @@
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              preTitle: Teambesprechung vom 23.05.
            </h1>
          </header>
-         <p
-           class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
-         >
-           Lead
-         </p>
        </div>
      </a>
    </article>
  </DocumentFragment>"
`;

exports[`Teaser Custom should render WithoutImage 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -9,20 +9,10 @@
        href="https://example.com"
      >
        <div
          class="css-1xecuhw-TeaserInnerWrapper e18of27r3"
        >
-         <img
-           alt="An image description"
-           class="css-17etez2-ImageWrapper-Teaser e1shv2yg0"
-           loading="lazy"
-           srcset="https://unsplash.it/800/800 800w,
- https://unsplash.it/500/500 500w,
- https://unsplash.it/300/300 300w,
- https://unsplash.it/200/200 200w"
-           title="An image title"
-         />
          <header
            class="css-4se2pd-TeaserTitle e18of27r2"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root""
`;

exports[`Teaser Custom should render WithoutPreTitle 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -25,11 +25,11 @@
            class="css-4se2pd-TeaserTitle e18of27r2"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
-             preTitle: Teambesprechung vom 23.05.
+             Teambesprechung vom 23.05.
            </h1>
          </header>
          <p
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >"
`;

exports[`Teaser Custom should render WithoutTitle 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -25,11 +25,11 @@
            class="css-4se2pd-TeaserTitle e18of27r2"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
-             preTitle: Teambesprechung vom 23.05.
+             preTitle: 
            </h1>
          </header>
          <p
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >"
`;

exports[`Teaser Event should render Default 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Teaser Event should render WithLightStyle 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -34,14 +34,9 @@
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title: Title
            </h1>
          </header>
-         <p
-           class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
-         >
-           Lead
-         </p>
        </div>
      </a>
    </article>
  </DocumentFragment>"
`;

exports[`Teaser Event should render WithoutDescription 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -37,11 +37,11 @@
            </h1>
          </header>
          <p
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
-           Lead
+           Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo cons
          </p>
        </div>
      </a>
    </article>
  </DocumentFragment>"
`;

exports[`Teaser Event should render WithoutDescriptionWithLightStyle 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -34,14 +34,9 @@
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title: Title
            </h1>
          </header>
-         <p
-           class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
-         >
-           Lead
-         </p>
        </div>
      </a>
    </article>
  </DocumentFragment>"
`;

exports[`Teaser Event should render WithoutImage 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Teaser Event should render WithoutImageWithoutEventImage 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -9,20 +9,10 @@
        href="https://example.com"
      >
        <div
          class="css-1xecuhw-TeaserInnerWrapper e18of27r3"
        >
-         <img
-           alt="An image description"
-           class="css-17etez2-ImageWrapper-Teaser e1shv2yg0"
-           loading="lazy"
-           srcset="https://unsplash.it/800/800 800w,
- https://unsplash.it/500/500 500w,
- https://unsplash.it/300/300 300w,
- https://unsplash.it/200/200 200w"
-           title="An image title"
-         />
          <time
            class="css-3uu3f-TeaserDate e18of27r1"
            datetime="2023-01-01T00:00:00.000Z"
          >
            01.01.2023"
`;

exports[`Teaser Event should render WithoutPreTitle 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -31,11 +31,11 @@
            class="css-4se2pd-TeaserTitle e18of27r2"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
-             Pre Title: Title
+             Title
            </h1>
          </header>
          <p
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >"
`;

exports[`Teaser Event should render WithoutTitle 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -31,11 +31,11 @@
            class="css-4se2pd-TeaserTitle e18of27r2"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
-             Pre Title: Title
+             Pre Title: Name of Event
            </h1>
          </header>
          <p
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >"
`;

exports[`Teaser Page should render Default 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Teaser Page should render WithLightStyle 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -28,14 +28,9 @@
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title: Title
            </h1>
          </header>
-         <p
-           class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
-         >
-           Lead
-         </p>
        </div>
      </a>
    </article>
  </DocumentFragment>"
`;

exports[`Teaser Page should render WithoutBlockWithLightStyle 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -28,14 +28,9 @@
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title: Title
            </h1>
          </header>
-         <p
-           class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
-         >
-           Lead
-         </p>
        </div>
      </a>
    </article>
  </DocumentFragment>"
`;

exports[`Teaser Page should render WithoutImage 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Teaser Page should render WithoutImageWithoutBlock 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Teaser Page should render WithoutLead 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -31,11 +31,11 @@
            </h1>
          </header>
          <p
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
-           Lead
+           Lead from block
          </p>
        </div>
      </a>
    </article>
  </DocumentFragment>"
`;

exports[`Teaser Page should render WithoutPreTitle 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -25,11 +25,11 @@
            class="css-4se2pd-TeaserTitle e18of27r2"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
-             Pre Title: Title
+             Title
            </h1>
          </header>
          <p
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >"
`;

exports[`Teaser Page should render WithoutTitle 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -25,11 +25,11 @@
            class="css-4se2pd-TeaserTitle e18of27r2"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
-             Pre Title: Title
+             Pre Title: Title from block
            </h1>
          </header>
          <p
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >"
`;

exports[`Teaser PeerArticle should render Default 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Teaser PeerArticle should render WithLightStyle 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -28,14 +28,9 @@
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title: Title
            </h1>
          </header>
-         <p
-           class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
-         >
-           Lead
-         </p>
        </div>
      </a>
    </article>
  </DocumentFragment>"
`;

exports[`Teaser PeerArticle should render WithoutAuthors 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Teaser PeerArticle should render WithoutBlockWithLightStyle 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -28,14 +28,9 @@
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title: Title
            </h1>
          </header>
-         <p
-           class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
-         >
-           Lead
-         </p>
        </div>
      </a>
    </article>
  </DocumentFragment>"
`;

exports[`Teaser PeerArticle should render WithoutDate 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Teaser PeerArticle should render WithoutImage 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Teaser PeerArticle should render WithoutImageWithoutBlock 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -9,20 +9,10 @@
        href="https://example.com"
      >
        <div
          class="css-1xecuhw-TeaserInnerWrapper e18of27r3"
        >
-         <img
-           alt="An image description"
-           class="css-17etez2-ImageWrapper-Teaser e1shv2yg0"
-           loading="lazy"
-           srcset="https://unsplash.it/800/800 800w,
- https://unsplash.it/500/500 500w,
- https://unsplash.it/300/300 300w,
- https://unsplash.it/200/200 200w"
-           title="An image title"
-         />
          <header
            class="css-4se2pd-TeaserTitle e18of27r2"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root""
`;

exports[`Teaser PeerArticle should render WithoutPreTitle 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -25,11 +25,11 @@
            class="css-4se2pd-TeaserTitle e18of27r2"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
-             Pre Title: Title
+             Pre Title on the article: Title
            </h1>
          </header>
          <p
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >"
`;

exports[`Teaser PeerArticle should render WithoutTitle 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -25,11 +25,11 @@
            class="css-4se2pd-TeaserTitle e18of27r2"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
-             Pre Title: Title
+             Pre Title: Title from block
            </h1>
          </header>
          <p
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >"
`;

exports[`Teaser should render WithClassName 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: grid;
}

@media (min-width:900px) {
  .emotion-0 {
    grid-column-start: 1;
    grid-column-end: 4;
    grid-row-start: 1;
    grid-row-end: 5;
  }
}

.emotion-2 {
  margin: 0;
  font: inherit;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  display: grid;
}

.emotion-3 {
  display: grid;
  gap: 8px;
  grid-template-rows: auto;
  grid-auto-rows: max-content;
}

.emotion-5 {
  max-width: 100%;
  width: 100%;
  height: auto;
  aspect-ratio: auto 1;
  height: 100%;
  width: 100%;
  object-fit: cover;
}

.emotion-9 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.334;
  letter-spacing: 0em;
}

.emotion-10 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 1.25rem;
  line-height: 1.6;
  letter-spacing: 0.0075em;
}

<article
    class="emotion-0 emotion-1"
    x="0"
    y="0"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone emotion-2"
      href="https://example.com"
    >
      <div
        class="extra-classname emotion-3 emotion-4"
      >
        <img
          alt="An image description"
          class="emotion-5 emotion-6"
          loading="lazy"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title="An image title"
        />
        <header
          class="emotion-7 emotion-8"
        >
          <h1
            class="MuiTypography-root MuiTypography-h5 emotion-9"
          >
            preTitle: Teambesprechung vom 23.05.
          </h1>
        </header>
        <p
          class="MuiTypography-root MuiTypography-h6 emotion-10"
        >
          Lead
        </p>
      </div>
    </a>
  </article>
</DocumentFragment>
`;

exports[`Teaser should render WithEmotion 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: grid;
}

@media (min-width:900px) {
  .emotion-0 {
    grid-column-start: 1;
    grid-column-end: 4;
    grid-row-start: 1;
    grid-row-end: 5;
  }
}

.emotion-2 {
  margin: 0;
  font: inherit;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  display: grid;
}

.emotion-3 {
  display: grid;
  gap: 8px;
  grid-template-rows: auto;
  grid-auto-rows: max-content;
}

.emotion-5 {
  max-width: 100%;
  width: 100%;
  height: auto;
  aspect-ratio: auto 1;
  height: 100%;
  width: 100%;
  object-fit: cover;
}

.emotion-9 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.334;
  letter-spacing: 0em;
}

.emotion-10 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 1.25rem;
  line-height: 1.6;
  letter-spacing: 0.0075em;
}

<article
    class="emotion-0 emotion-1"
    x="0"
    y="0"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone emotion-2"
      href="https://example.com"
    >
      <div
        class="emotion-3 emotion-4"
      >
        <img
          alt="An image description"
          class="emotion-5 emotion-6"
          loading="lazy"
          srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
          title="An image title"
        />
        <header
          class="emotion-7 emotion-8"
        >
          <h1
            class="MuiTypography-root MuiTypography-h5 emotion-9"
          >
            preTitle: Teambesprechung vom 23.05.
          </h1>
        </header>
        <p
          class="MuiTypography-root MuiTypography-h6 emotion-10"
        >
          Lead
        </p>
      </div>
    </a>
  </article>
</DocumentFragment>
`;
